compiler/parser/chapel.lex

```diff
@@ -425,36 +445,36 @@ static const char* eatStringLiteral(yyscan_t scanner, const char* startChar) {
       yyerror(yyLloc, &context, "end-of-line in a string literal without a preceding backslash");
     } else {
       if (startCh == '\'' && c == '\"') {
-        addCharEscape('\\');
+        addCharEscapeNonprint('\\');
       }
 
       // \ escape ? to avoid C trigraphs
       if (c == '?')
-        addCharEscape('\\');
+        addCharEscapeNonprint('\\');
 
-      addCharEscape(c);
+      addCharEscapeNonprint(c);
     }
 
     if (c == '\\') {
       c = getNextYYChar(scanner);
 
       if (c == '\n') {
         processNewline(scanner);
-        addCharEscape('n');
+        addCharEscapeNonprint('n');
       } else if (c == 'u' || c == 'U') {
         ParserContext context(scanner);
         yyerror(yyLloc, &context, "universal character name not yet supported in string literal");
-        addCharEscape('t'); // add a valid escape to continue parsing
+        addCharEscapeNonprint('t'); // add a valid escape to continue parsing
       } else if ('0' <= c && c <= '7' ) {
         ParserContext context(scanner);
         yyerror(yyLloc, &context, "octal escape not supported in string literal");
-        addCharEscape('t'); // add a valid escape to continue parsing
+        addCharEscapeNonprint('t'); // add a valid escape to continue parsing
       } else if (c == 0) {
         // we've reached EOF
-        addCharEscape('t'); // add a valid escape to continue parsing
+        addCharEscapeNonprint('t'); // add a valid escape to continue parsing
```

Comment:
I think any of these with a known character should just use addChar, but that's not strictly required for your change.